src/constructUrl.js

```diff
@@ -116,28 +116,39 @@ function constructUrl(src, longOptions) {
   if (!src) {
     return "";
   }
+  const params = compactParamKeys(longOptions);
+  const { client, pathComponents } = extractClientAndPathComponents(src);
+  return client.buildURL(pathComponents.join("/"), params);
+}
 
+function compactParamKeys(longOptions) {
   const keys = Object.keys(longOptions);
   const keysLength = keys.length;
-  let url = src + "?";
+  const params = {};
   for (let i = 0; i < keysLength; i++) {
     let key = keys[i];
-    let val = longOptions[key];
-
     if (PARAM_EXPANSION[key]) {
-      key = PARAM_EXPANSION[key];
+      params[PARAM_EXPANSION[key]] = longOptions[key];
     } else {
-      key = encodeURIComponent(key);
+      params[key] = longOptions[key];
     }
+  }
 
-    if (key.substr(-2) === "64") {
-      val = Base64.encodeURI(val);
-    }
+  return params;
+}
 
-    url += key + "=" + encodeURIComponent(val) + "&";
-  }
+function extractClientAndPathComponents(src) {
+  const [scheme, rest] = src.split("://");
+  const [domain, ...pathComponents] = rest.split("/");
+  let useHTTPS = scheme == "https";
+
+  const client = new ImgixClient({
+    domain: domain,
+    useHTTPS: useHTTPS,
+    includeLibraryParam: false,
```

Comment:
Is `ixlib` reset somewhere else?

Reply:
Yes, the client's `ixlib` is disabled here in order to defer to react-imgix's way of setting this value.
https://github.com/imgix/react-imgix/blob/ae9bcfb773dcb92f53319cea01a1c02ba8cb51bc/src/react-imgix.jsx#L101